src/react/account/AccountRoleSelectModal.tsx

```diff
@@ -0,0 +1,167 @@
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useRouteMatch } from 'react-router';
+import Table from '@scality/core-ui/dist/components/tablev2/Tablev2.component';
+import { Button } from '@scality/core-ui/dist/next';
+import { Tooltip } from '@scality/core-ui';
+import { AppState } from '../../types/state';
+import { closeAccountRoleSelectModal } from '../actions';
+import { CustomModal as Modal, ModalBody } from '../ui-elements/Modal';
+import { regexArn, SCALITY_INTERNAL_ROLES, useAccounts } from '../utils/hooks';
+import {
+  useCurrentAccount,
+  useDataServiceRole,
+} from '../DataServiceRoleProvider';
+import { getRoleArnStored, setRoleArnStored } from '../utils/localStorage';
+import { Icon } from '../ui-elements/Help';
+
+function AccountRoleSelectModal() {
+  const dispatch = useDispatch();
+  const accounts = useAccounts();
+  const { path } = useRouteMatch();
+
+  const { account, selectAccountAndRoleRedirectTo } = useCurrentAccount();
+  const { roleArn } = useDataServiceRole();
+  const [assumedRoleArn, setAssumedRoleArn] = useState(roleArn);
+  const [currentAccountName, setCurrentAccountName] = useState(account?.Name);
+  const isModalOpen = useSelector(
+    (state: AppState) => state.uiAccounts.showAccountRoleSelect,
+  );
+
+  const accountsWithRoles = [];
+  accounts.forEach((account) => {
+    const accountName = account.Name;
+    account.Roles.forEach((role) => {
+      const rolePathName = regexArn.exec(role.Arn)?.groups['role_name'] || '';
+      const rolePath = rolePathName.includes('/')
+        ? rolePathName.split('/')[0]
+        : '/';
+      const roleName = rolePathName.split('/')[1] || rolePathName;
+      accountsWithRoles.push({
+        accountName,
+        roleName,
+        rolePath,
+        roleArn: role.Arn,
+      });
+    });
+  });
+
+  if (!isModalOpen) {
+    return null;
+  }
+
+  const handleClose = () => {
+    dispatch(closeAccountRoleSelectModal());
+  };
```

Comment:
I think we could simplify a bit this and remove the redux logic by collocating the modal with it's opening button (I mean within the same component). Then we wouldn't need redux, just render the ButtonWithModal component and we get the whole logic for free

Reply:
This ButtonWithModal component could simply be the `AccountSelectorButton`